linode_api4/groups/lke.py

```diff
@@ -107,41 +107,22 @@ def cluster_create(
         :returns: The new LKE Cluster
         :rtype: LKECluster
         """
-        pools = []
-        if not isinstance(node_pools, list):
-            node_pools = [node_pools]
-
-        for c in node_pools:
-            if isinstance(c, dict):
-                new_pool = {
-                    "type": (
-                        c["type"].id
-                        if "type" in c and issubclass(type(c["type"]), Base)
-                        else c.get("type")
-                    ),
-                    "count": c.get("count"),
-                }
-
-                pools += [new_pool]
 
         params = {
             "label": label,
-            "region": region.id if issubclass(type(region), Base) else region,
-            "node_pools": pools,
-            "k8s_version": (
-                kube_version.id
-                if issubclass(type(kube_version), Base)
-                else kube_version
-            ),
-            "control_plane": (
-                control_plane.dict
-                if issubclass(type(control_plane), JSONObject)
-                else control_plane
+            "region": region,
+            "k8s_version": kube_version,
+            "node_pools": (
+                node_pools if isinstance(node_pools, list) else [node_pools]
             ),
+            "control_plane": control_plane,
         }
         params.update(kwargs)
 
-        result = self.client.post("/lke/clusters", data=drop_null_keys(params))
+        result = self.client.post(
+            "/lke/clusters",
+            data=_flatten_request_body_recursive(drop_null_keys(params)),
```

Comment:
We might be able to apply the same refactor in other group-level endpoint functions 👀